test/com/amazon/ion/streaming/ReaderTest.java

```diff
@@ -284,10 +284,10 @@ public void testReadingDecimalAsLong()
         assertEquals(readLong("9223372036854775807."), readLong("9223372036854775807")); // Max long
     }
 
-    @Test(expected = IllegalStateException.class)
```

Comment:
I'm sort of agnostic about why the new code is preferable.  Nothing really strikes me as wrong about the old way, but then I think I've been asked to make a similar change on a code base long ago and far away so IDK.

Reply:
The new version is more tightly specified- it pushes the test towards declaring exactly which statement should throw, and implicitly disallows exceptions elsewhere. It also allows us to make further assertions about the exception, if we want to.